baybe/utils/validation.py

```diff
@@ -68,3 +73,98 @@ def validator(self: Any, attribute: Attribute, value: Any) -> None:
 
 non_inf_float = _make_restricted_float_validator(allow_nan=True, allow_inf=False)
 """Validator for non-infinite floats."""
+
+
+def validate_target_input(data: pd.DataFrame, targets: Iterable[Target]) -> None:
+    """Validate input dataframe columns corresponding to targets.
+
+    Args:
+        data: The input dataframe to be validated.
+        targets: The allowed targets.
+
+    Raises:
+        ValueError: If the input dataframe is empty.
+        ValueError: If any target data contain NaN.
+        TypeError: If any numerical target data contain non-numeric values.
+        ValueError: If any binary target data contain values not part of the targets'
+            allowed values.
+    """
+    from baybe.targets import BinaryTarget, NumericalTarget
+
+    if len(data) < 1:
```

Comment:
I think is slightly better since otherwise the code would continue in case of a degenerate zero-column dataframe
```suggestion
    if data.empty:
```

Reply:
changed